### PR TITLE
Link to SPDX license list in linter message

### PR DESCRIPTION
**Pull request by jakirkham**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -660,7 +660,9 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
         expected_exceptions = set([l.strip() for l in expected_exceptions])
     if set(filtered_licenses) - expected_licenses:
         hints.append(
```
Comment (beckermr) [inline conda_smithy/lint_recipe.py:662]: To require spdx we would need to make this a lint.

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -660,7 +660,9 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
-            "License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression."
+            "License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.\n\n"
+            "A full list of acceptable licenses can be found "
+            "[here]( https://spdx.org/licenses/ )."
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:665]: I think we should link to out documentation at https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions

File: conda_smithy/lint_recipe.py (hunk 1 above)
Comment (jakirkham) [reply conda_smithy/lint_recipe.py:662]: Sorry I'm not trying to change hint/lint. Just trying to add the link. I'll update the note accordingly.

File: conda_smithy/lint_recipe.py (hunk 2 above)
Comment (jakirkham) [reply conda_smithy/lint_recipe.py]: Thanks! Changed :)

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -660,7 +660,9 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
         expected_exceptions = set([l.strip() for l in expected_exceptions])
     if set(filtered_licenses) - expected_licenses:
         hints.append(
-            "License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression."
+            "License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.\n\n"
+            "A full list of acceptable licenses can be found "
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:664]: The link doesn't give a full list, but a few examples and links to the full list.
```suggestion
            "Documentation can be found "
```

Comment (jakirkham) [reply conda_smithy/lint_recipe.py]: Good point. I made a similar change below. Please take a look.

**Comment by scopatz** [comment]
Thanks @jakirkham! This is failing CI

**Comment by jakirkham** [comment]
Thanks. Have fixed the test. Also added this note to the other SPDX linter failure case.

Please let me know if there's anything else 🙂

**Pull request merged**